src/transports/tcp/atcp.c

```diff
@@ -192,6 +201,11 @@ static void nn_atcp_handler (struct nn_fsm *self, int src, int type,
                 nn_assert (sz == sizeof (val));
                 nn_usock_setsockopt (&atcp->usock, SOL_SOCKET, SO_RCVBUF,
                     &val, sizeof (val));
+                sz = sizeof (val);
+                nn_ep_getopt (atcp->ep, NN_TCP, NN_TCP_NODELAY, &val, &sz);
+                nn_assert (sz == sizeof (val));
+                nn_usock_setsockopt (&atcp->usock, IPPROTO_TCP, SO_RCVBUF,
```

Comment:
This is incorrect.  It is setting the SO_RCVBUF, not the TCP_NODELAY option.   I think you made a copy-pasta error.